lib/segment/src/data_types/build_index_result.rs

```diff
@@ -0,0 +1,16 @@
+use crate::index::field_index::FieldIndex;
+use crate::types::PayloadFieldSchema;
+
+pub enum BuildFieldIndexResult {
```

Comment:
This result structure enforces the caller to handle each case, including the `IncompatibleSchema`. It makes sure we don't forget to follow the protocol of removing existing index (otherwise it is impossible to handle `IncompatibleSchema` case)